### PR TITLE
Update/multi block mover

### LINKED ISSUE #5752: Block movers: wrong aria-label info with blocks multi-selection
Noticed while investigating on #5751 

To reproduce:
- add 3 paragraphs and some more blocks after them
- select the 3 first paragraphs, e.g.:

screen shot 2018-03-22 at 23 01 56

- check the aria-label text of the "move up" button

The text is `Move 3 blocks from position 1 up by one place` which is not correct: there's no position before "1". Same when in the last position, the text will say that it is still possible to move down by 1 position.

### PR DESCRIPTION
**Pull request by dom-o**
## Description
Fixes #5752 

## How Has This Been Tested?
Manually tested;  didn't find any regressions.
Ran npm test and all still pass.

move_top_up

move_bottom_down

## Types of changes
Multi-selected block mover buttons are disabled when blocks are at the top/bottom; aria-labels are set accordingly.

### CODE REVIEW

**Review by aduth — CHANGES_REQUESTED** [review]
`BlockListBlock`, the parent component which renders this one, already has `isFirst` and `isLast` props. It also renders a `<BlockMover />` of its own for singular selection, and `<BlockListMultiControls />` for a multi-selection. Seems we could just pass those from there instead of calculating it again here?

https://github.com/WordPress/gutenberg/blob/3e56452edb142e5bb72cf5bf025b7afe9d441e78/editor/components/block-list/block.js#L408-L409

https://github.com/WordPress/gutenberg/blob/3e56452edb142e5bb72cf5bf025b7afe9d441e78/editor/components/block-list/block.js#L518

**Edit:** Actually, this will work for `isFirst`, but since the movers are only rendered for the first block in a multi-selection, it won't know if the extent of its multi-selection is the end of the block list. So we may still need `connect` to provide this information. It could be provided as `isLast` direct from `connect` (where `lastIndex + 1 === blockCount` occurs in the `connect` function).

...or, maybe the fact that the source of these props would become disjointed this way could be good justification for the approach as you have proposed, since it's all relatively well-contained, even if we are repeating some information already available from the direct parent of the multi-controls.

Thoughts? 😄

**Comment by dom-o** [comment]
I originally looked into passing from `BlockListBlock`, but the separate sources did feel a little weird. I'm not married to it either way; both would work. Providing them straight from `connect `is good though.

File: editor/components/block-list/multi-controls.js [hunk 1]
```diff
@@ -33,9 +38,17 @@ function BlockListMultiControls( { multiSelectedBlockUids, rootUID, isSelecting
 	];
 }
 
-export default connect( ( state ) => {
+export default connect( ( state, ownProps ) => {
+	const { rootUID } = ownProps;
+	const uids = getMultiSelectedBlockUids( state );
+
+	const firstIndex = getBlockIndex( state, first( uids ), rootUID );
```
Comment (aduth) [inline editor/components/block-list/multi-controls.js]: Alternatively, we could use `getFirstMultiSelectedBlockUid` and `getLastMultiSelectedBlockUid`, but this isn't very complex logic, so seems reasonable enough.

**Review by aduth — APPROVED** [review]
Looks great, thanks 👍 

Aside: It's more of an issue with nested blocks and their layouts (one which is not easy to solve), but down/up arrow will still be shown for multi-selection of a complete first/last column.

image

**Pull request merged**